ShopNow/UserApp/views.py

```diff
@@ -1,134 +1,402 @@
             else: 
-                return Response(serialized.errors)      
-             
+                return Response(serialized.errors, status = status.HTTP_400_BAD_REQUEST)      
 
+class RegisteredUsers(APIView):
+    authentication_classes = [JWTAuthentication]
+    permission_classes = [CustomizeAPIPermissions]
 
-class get_register_users(APIView):
-    authentication_classes=[JWTAuthentication]
-    permission_classes=[CustomizeAPIPermissions]
+    def get(self, request, pk = None):
+            if pk is not None:
+                try:
+                    instance = Webuser.objects.get(id = pk)
+                except:
+                    return Response("User Not exists ", status = status.HTTP_404_NOT_FOUND)
+                self.check_object_permissions(request, instance)
+                serialized = WebUserSerializer(instance)
+                return Response(serialized.data) 
+            user = Webuser.objects.all()
+            serialized = WebUserSerializer(user, many = True)
+            return Response(serialized.data)
+           
+    def put(self, request, pk = None):
+        if pk is None:
+           return Response({"error": "You are not adding the pk"}, status = status.HTTP_400_BAD_REQUEST)
+        instance = Webuser.objects.get(id = pk)
+        self.check_object_permissions(request, instance)
+        data = request.data
+        serialized = WebUserSerializer(instance, data = data)
+        if not serialized.is_valid():
+            return Response(serialized.errors, status = status.HTTP_400_BAD_REQUEST)   
+        serialized.save()
+        return Response(serialized.data)                   
 
-    def get(self,request,pk=None):
+    def patch(self, request, pk = None):
         if pk is None:
-            user=Webuser.objects.all()
-            serialized=WebUserSerializer(user,many=True)
-            print("This is user: ",request.user)
-            return Response({'user':serialized.data})
+            return Response({"error": "You are not adding the pk"}, status = status.HTTP_400_BAD_REQUEST)
+        instance=Webuser.objects.get(id = pk)
+        self.check_object_permissions(request, instance)
+        data = request.data
+        serialized = WebUserSerializer(instance, data = data, partial = True)
+        if not serialized.is_valid():
+            return Response(serialized.errors, status = status.HTTP_400_BAD_REQUEST)   
+        serialized.save()
+        return Response(serialized.data)   
+
+    def delete(self, request, pk = None):
+            if pk is None:
+                return Response({"error": "You are not adding the pk"}, status = status.HTTP_400_BAD_REQUEST)
+            instance = Webuser.objects.get(id = pk)
+            self.check_object_permissions(request, instance)
+            instance.delete()
+            return Response({'out':"Deleted successfully!!"})
+
+class UserAdminFrontend(View): 
```

Comment:
This looks like frontend code, meaning this we are consuming our API, we should make a separate view for this and keep the backend and frontend logic separate. 

You should make a separate folder like the name 'frontend' and keep all these things frontend view. I can elaborate this in checking meeting if you have any confusion.

frontend/           # Folder for traditional template views
│   ├── views.py
│   ├── templates/